lib/lte_link_control/lte_lc.c

```diff
@@ -151,32 +182,328 @@ static const char cgauth[] = "AT+CGAUTH="CONFIG_LTE_PDN_AUTH;
 static const char legacy_pco[] = "AT%XEPCO=0";
 #endif
 
-void at_handler(void *context, const char *response)
+enum lte_lc_notif_type {
+	LTE_LC_NOTIF_CEREG,
+	LTE_LC_NOTIF_CSCON,
+	LTE_LC_NOTIF_CEDRXP,
+
+	LTE_LC_NOTIF_COUNT,
+};
+
+static const char *const at_notifs[] = {
+	[LTE_LC_NOTIF_CEREG] = "+CEREG",
+	[LTE_LC_NOTIF_CSCON] = "+CSCON",
+	[LTE_LC_NOTIF_CEDRXP] = "+CEDRXP",
+};
+
+BUILD_ASSERT(ARRAY_SIZE(at_notifs) == LTE_LC_NOTIF_COUNT);
+
+static bool is_relevant_notif(const char *notif, enum lte_lc_notif_type *type)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(at_notifs); i++) {
+		if (strncmp(at_notifs[i], notif,
+			    strlen(at_notifs[i])) == 0) {
+			/* The notification type matches the array index */
+			*type = i;
+
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static int parse_cereg(const char *notification,
+		       enum lte_lc_nw_reg_status *reg_status,
+		       struct lte_lc_cell *cell,
+		       struct lte_lc_psm_cfg *psm_cfg)
+{
+	int err, status;
+	struct at_param_list resp_list = {0};
+	char str_buf[10];
+	size_t len = sizeof(str_buf) - 1;
+
+	if (notification == NULL) {
```

Comment:
This is a static function. At this point, you should not check the validity of input parameters. This should be done when calling the function.
Edit: You already have a check for this in the `at_handler` function.